### PR TITLE
Updates current `project_root()`

### LINKED ISSUE #46: Fix utils/project_root() to be Atomkraft-specific
Current implementation of [project_root() function](https://github.com/informalsystems/atomkraft/blob/6f1cb56f37bd57526a87386125362eff1034896b/atomkraft/utils/project.py#L5-L11):

```python
def project_root():
    cwd = Path(os.getcwd())
    while cwd != cwd.parent:
        if (cwd / "pyproject.toml").exists():
            return cwd
        cwd = cwd.parent
    return None
```
 is dangerous: it traverses up to whatever "pyproject.toml" it finds, and returns this directory. There may be other Poetry projects up there, which are not Atomkraft projects. As a result, arbitrary unrelated projects could be overwritten.

Instead, this function should search for Atomkraft-specific config files, e.g. `.atomkraft` directory.

### PR DESCRIPTION
**Pull request by rnbguy**
Fixes #46

### CODE REVIEW

File: atomkraft/test/trace.py [hunk 1]
```diff
@@ -32,11 +32,8 @@ def test_trace(trace: PathLike, reactor: PathLike, keypath: str):
             "could not find Atomkraft project: are you in the right directory?"
         )
 
-    tests = os.path.join(root, "tests")
-    if not os.path.isdir(tests):
-        raise RuntimeError(
-            "Atomkraft tests directory not found: have you executed `atomkraft init`?"
-        )
+    tests = root / "tests"
+    tests.mkdir(exist_ok=True)
```
Comment (rnbguy) [inline atomkraft/test/trace.py:36]: I propose to make `tests` directory optional for a project. Create it on-the-fly when needed - as we use the traces directly via `atomkraft test traces`.

File: atomkraft/utils/project.py [hunk 2]
```diff
@@ -5,7 +5,10 @@
 def project_root():
     cwd = Path(os.getcwd())
     while cwd != cwd.parent:
-        if (cwd / "pyproject.toml").exists():
+        if (
+            (cwd / "pyproject.toml").exists()
+            and (cwd / "atomkraft.toml").exists()
+            and (cwd / ".atomkraft" / "config.toml").exists()
```
Comment (rnbguy) [inline atomkraft/utils/project.py:11]: Project root is the nearest parent with `./pyproject.toml`, `./atomkraft.toml`, `./.atomkraft/config.toml` present.

File: template/atomkraft.toml [hunk 3]
```diff
@@ -1,8 +1,2 @@
-[chain]
-binary = "gaia"
-
-[modelator.apalache]
-jar = "/path/to/apalache/jar"
-
-[tests]
-lightweight = ["authz", "cosmwasm"]
+[project]
+chain-file = "chain.toml"
```
Comment (rnbguy) [inline template/atomkraft.toml:2]: Updated the config to look sensible, even though we don't use it.

**Review by ivan-gavran — APPROVED** [review]
This looks good!

**Review by hvanz — APPROVED** [review]
LGTM

File: atomkraft/test/trace.py (hunk 1 above)
Comment (andrey-kuprianov) [reply atomkraft/test/trace.py:36]: As a short-term solution this is fine. As a long-term, we should automate the directory management similar to the [config file management](https://github.com/informalsystems/atomkraft/tree/dev/atomkraft/config).

File: atomkraft/utils/project.py (hunk 2 above)
Comment (andrey-kuprianov) [reply atomkraft/utils/project.py:11]: looks good

**Review by andrey-kuprianov — APPROVED** [review]
LGTM!

**Pull request merged**